cmd/swarmctl/service/flagparser/restart.go

```diff
@@ -10,25 +10,26 @@ import (
 )
 
 func parseRestart(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
-	// parse restart-condition
-	condition, err := flags.GetString("restart-condition")
-	if err != nil {
-		return err
-	}
+	if spec.Task.Restart == nil || flags.Changed("restart-condition") {
```

Comment:
How about this? I'm not sure tangling it with `--restart-condition` brings any benefit. Or even create `spec.Task.Restart` at task creation. 

```
		if spec.Task.Restart == nil {
			spec.Task.Restart = &api.RestartPolicy{}
		}

	if flags.Changed("restart-condition") {
		condition, err := flags.GetString("restart-condition")
		if err != nil {
			return err
		}

		switch condition {
		case "none":
			spec.Task.Restart.Condition = api.RestartOnNone
		case "failure":
			spec.Task.Restart.Condition = api.RestartOnFailure
		case "any":
			spec.Task.Restart.Condition = api.RestartOnAny
		default:
			return fmt.Errorf("invalid restart condition: %s", condition)
		}
	}
```